### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by ErikRehmTT**
- integration tests run and finish sucessful
- add integration test gradle task
- integration tests run on internal jenkins builds

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -66,16 +65,22 @@ dependencies {
...
     testImplementation 'org.jenkins-ci.plugins.workflow:workflow-step-api:625.vd896b_f445a_f8:tests'
     testImplementation "org.jenkins-ci.main:jenkins-war:${jenkinsVersion}"
-    testImplementation('org.testcontainers:spock:1.17.3') {
+    testImplementation('org.testcontainers:spock:1.17.4') {
         exclude group: 'org.spockframework', module: 'spock-core'
     }
-    testImplementation 'org.spockframework:spock-core:2.1-groovy-3.0'
+    testImplementation 'org.spockframework:spock-core:2.3-groovy-3.0'
+    testImplementation 'org.spockframework:spock-junit4:2.3-groovy-3.0'
     testImplementation 'junit:junit:4.13.2'
```
Comment (MartinGroscheTT) [inline build.gradle:73]: There are two junit testimplementations? What about cross-effects?

File: build.gradle [hunk 2]
```diff
@@ -164,14 +169,31 @@ tasks.named('server').configure {
...
-        excludeTestsMatching '*IT' // TODO fix issue with Jenkins ITs
+        excludeTestsMatching '*IT'
     }
     useJUnitPlatform()
     finalizedBy jacocoTestReport
 }
 
+task integrationTest(type: Test) {
+    // exclude code, which can not be instrumented correctly
+    jacoco {
```
Comment (MartinGroscheTT) [inline build.gradle:187]: Avoid code duplication with [custom tasks](https://docs.gradle.org/current/userguide/custom_tasks.html)

File: gradle.properties [hunk 3]
```diff
@@ -7,5 +7,4 @@ openapiVersion=6.1.0
 jenkinsTestHarnessVersion=1746.v2b_3c5d1983b_e
 # --- Disabling SHA256 and SHA512 checksums when releasing a plugin ---#
 # --- see https://github.com/jenkinsci/gradle-jpi-plugin/blob/master/README.md#disabling-sha256-and-sha512-checksums-when-releasing-a-plugin ---#
-org.gradle.internal.publish.checksums.insecure=true
-
+org.gradle.internal.publish.checksums.insecure=true
```
Comment (MartinGroscheTT) [inline gradle.properties:10]: Add new line at the end

**Review by MartinGroscheTT — CHANGES_REQUESTED** [review]
delete [containerTest.properties](https://github.com/jenkinsci/ecu-test-execution-plugin/pull/10/files#diff-3d103fc7c312a3e136f88e81cef592424b8af2464c468116545c4d22d6edcf19). File is part of the .gitignore

File: build.gradle (hunk 1 above)
Comment (ErikRehmTT) [reply build.gradle:73]: spock-core 2.x does not support junit 4 natively anymore. Therefore they created the spock-junit4 package which provides a compat layer.

File: build.gradle [hunk 4]
```diff
@@ -69,7 +69,7 @@ dependencies {
         exclude group: 'org.spockframework', module: 'spock-core'
     }
     testImplementation 'org.spockframework:spock-core:2.3-groovy-3.0'
-    testImplementation 'org.spockframework:spock-junit4:2.3-groovy-3.0'
+    testImplementation 'org.spockframework:spock-junit4:2.3-groovy-3.0' // compat layer for junit4
     testImplementation 'junit:junit:4.13.2'
```
Comment (MartinGroscheTT) [inline build.gradle:73]: ok. I meant, why we use both of them. Imo we can delete the _testImplementation 'junit:junit:4.13.2'_

**Review by MartinGroscheTT — CHANGES_REQUESTED** [review]

**Review by MartinGroscheTT — APPROVED** [review]

**Pull request merged**